src/test/java/com/sap/sgs/phosphor/fosstars/data/github/HasSecurityPolicyTest.java

```diff
@@ -1,73 +1,91 @@
+    GitHubDataFetcher.instance().repositoryCache.clear();
+    GitHubDataFetcher.instance().commitsCache.clear();
   }
 
   @Test
-  public void testApacheNifi() throws IOException {
-    assumeNotNull(github);
+  public void testHasPolicy() throws IOException {
+    GHContent content = mock(GHContent.class);
+    when(content.isFile()).thenReturn(true);
+    when(content.getSize()).thenReturn(1000L);
+
+    GHRepository repository = mock(GHRepository.class);
+    when(repository.getFileContent("SECURITY.md")).thenReturn(content);
+
+    GitHub github = mock(GitHub.class);
+    when(github.getRepository(any())).thenReturn(repository);
+
+    GitHubDataFetcher fetcher = GitHubDataFetcher.instance();
+    fetcher.repositoryCache.clear();
+
     HasSecurityPolicy provider = new HasSecurityPolicy(github);
-    ValueHashSet values = new ValueHashSet();
-    GitHubProject project = new GitHubProject("apache", "nifi");
-    provider.set(new Terminal()).update(project, values);
-    assertTrue(values.has(SCANS_FOR_VULNERABLE_DEPENDENCIES));
-    Optional<Value> something = values.of(SCANS_FOR_VULNERABLE_DEPENDENCIES);
-    assertNotNull(something);
-    assertTrue(something.isPresent());
-    Value<Boolean> value = something.get();
-    assertNotNull(value);
-    assertEquals(Boolean.TRUE, value.get());
+    provider.set(new GitHubProjectValueCache());
+
+    check(provider, true);
   }
 
   @Test
-  public void testApacheCxf() throws IOException {
-    assumeNotNull(github);
-    GitHubProject project = new GitHubProject("apache", "cxf");
+  public void testNoPolicy() throws IOException {
+    GHRepository repository = mock(GHRepository.class);
+    when(repository.getFileContent("SECURITY.md")).thenThrow(IOException.class);
+
+    GitHub github = mock(GitHub.class);
+    when(github.getRepository(any())).thenReturn(repository);
+
+    GitHubDataFetcher fetcher = GitHubDataFetcher.instance();
+    fetcher.repositoryCache.clear();
+
     HasSecurityPolicy provider = new HasSecurityPolicy(github);
+    provider.set(new GitHubProjectValueCache());
+
+    check(provider, false);
+  }
+
+  private static void check(HasSecurityPolicy provider, boolean expectedValue) throws IOException {
     ValueHashSet values = new ValueHashSet();
-    provider.set(new Terminal()).update(project, values);
-    assertTrue(values.has(SCANS_FOR_VULNERABLE_DEPENDENCIES));
-    Optional<Value> something = values.of(SCANS_FOR_VULNERABLE_DEPENDENCIES);
+    GitHubProject project = new GitHubProject("org", "test");
+    provider.update(project, values);
+
+    assertEquals(1, values.size());
+    assertTrue(values.has(HAS_SECURITY_POLICY));
+    Optional<Value> something = values.of(HAS_SECURITY_POLICY);
     assertNotNull(something);
     assertTrue(something.isPresent());
     Value<Boolean> value = something.get();
     assertNotNull(value);
-    assertEquals(Boolean.FALSE, value.get());
+    assertEquals(expectedValue, value.get());
+  }
+
+  @AfterClass
```

Comment:
Can we have both @AfterClass and @Before annotated to one method?

Reply:
I've never thought about that. Let me try.

Reply:
It works. I merged those methods.